sycl/plugins/unified_runtime/ur/adapters/hip/enqueue.cpp

```diff
@@ -82,6 +82,61 @@ void simpleGuessLocalWorkSize(size_t *ThreadsPerBlock,
     --ThreadsPerBlock[0];
   }
 }
+
+ur_result_t setHipMemAdvise(const void *DevPtr, size_t Size,
+                            ur_usm_advice_flags_t URAdviceFlags,
+                            hipDevice_t Device) {
+  std::unordered_map<ur_usm_advice_flags_t, hipMemoryAdvise>
+      URToHIPMemAdviseDeviceFlagsMap = {
+          {UR_USM_ADVICE_FLAG_SET_READ_MOSTLY, hipMemAdviseSetReadMostly},
+          {UR_USM_ADVICE_FLAG_CLEAR_READ_MOSTLY, hipMemAdviseUnsetReadMostly},
+          {UR_USM_ADVICE_FLAG_SET_PREFERRED_LOCATION,
+           hipMemAdviseSetPreferredLocation},
+          {UR_USM_ADVICE_FLAG_CLEAR_PREFERRED_LOCATION,
+           hipMemAdviseUnsetPreferredLocation},
+          {UR_USM_ADVICE_FLAG_SET_ACCESSED_BY_DEVICE,
+           hipMemAdviseSetAccessedBy},
+          {UR_USM_ADVICE_FLAG_CLEAR_ACCESSED_BY_DEVICE,
+           hipMemAdviseUnsetAccessedBy},
+      };
+  for (auto &FlagPair : URToHIPMemAdviseDeviceFlagsMap) {
+    if (URAdviceFlags & FlagPair.first) {
+      UR_CHECK_ERROR(hipMemAdvise(DevPtr, Size, FlagPair.second, Device));
+    }
+  }
+
+  static std::unordered_map<ur_usm_advice_flags_t, hipMemoryAdvise>
```

Comment:
In the present use, this doesn't need the heavyweight as an unordered map.

You can cleanly get away with an array of pairs or custom structs (either C-style or `std::array`>

Reply:
Sure. That was a careless copy of similar mapping from code in the cuda adapter. I too agree the map isn't needed indeed.